backend/matching-service/app/src/matching/matching.schemas.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from "zod";
+import { HTTP_STATUS } from "../types/http";
+
+export const createRoomSchema = z.union([
+  z.object({
+    status: z.literal(HTTP_STATUS.SUCCESS),
+    data: z.object({
+      created: z.string(),
+      roomId: z.string(),
+    }),
+  }),
+  z.object({
+    status: z.literal(HTTP_STATUS.FAIL),
+    data: z.object({
+      message: z.string(),
+    }),
```

Comment:
```suggestion
    data: z.object({
      message: z.string(),
    }).optional(),
```

just in case there is no data or message returned

Reply:
6119054859773dc7ac52299017b295e195a9ed4a